openqaoa/optimizers/optimization_methods_pennylane.py

```diff
@@ -0,0 +1,160 @@
+    Read https://docs.pennylane.ai/en/stable/introduction/interfaces.html#optimizers
+
+    PARAMETERS
+    ----------
+    fun : callable
+        Function to minimize
+    x0 : ndarray
+        Initial guess.
+    args : sequence, optional
+        Arguments to pass to `func`.
+    maxfev : int, optional
+        Maximum number of function evaluations.
+    method : string, optional
+        Optimizer method to compute the steps.
+    qfim : callable, optional (required for natural_grad_descent)
+        Callable Fubini-Study metric tensor
+    maxiter : int, optional
+        Maximum number of iterations.
+    tol : float
+        Tolerance before the optimizer terminates; if `tol` is larger than the difference between two steps, terminate optimization.
+    jac : callable, optinal (required for all methods but rotosolve and spsa)
+        Callable gradient function.
+    callback : callable, optional
+        Called after each iteration, as ``callback(xk)``, where ``xk`` is the
+        current parameter vector.
+    options : dict, optional
+        Dictionary where keys are the arguments for the optimizers object, and
+        the values are the values to pass to these arguments.
+        To know all the possible argumets read
+        https://docs.pennylane.ai/en/stable/introduction/interfaces.html#optimizers.
+
+
+    (read https://docs.pennylane.ai/en/stable/code/api/pennylane.RotosolveOptimizer.html#pennylane.RotosolveOptimizer.step)
+    nums_frequency : dict[dict], required for rotosolve
+        The number of frequencies in the fun per parameter.
+    spectra : dict[dict], required for rotosolve
+        Frequency spectra in the objective_fn per parameter.
+    shifts : dict[dict], required for rotosolve
+        Shift angles for the reconstruction per parameter.
+
+
+    RETURNS
+    -------
+    OptimizeResult : OptimizeResult
+        Scipy OptimizeResult object.
+    '''
+
+    def cost(params, **k): # define a function to convert the params list from pennylane to numpy
+        return fun(np.array(params), *k)
+
+
+    optimizer = AVAILABLE_OPTIMIZERS[method] # define the optimizer
+
+    #get optimizer arguments
+    arguments = inspect.signature(optimizer).parameters.keys()
+    options_keys = list(options.keys())
+
+    #check which values of the options dict can be passed to the optimizer (pop the others)
+    for key in options_keys:
+        if key not in arguments: options.pop(key) 
+        if 'maxiter' in arguments: options['maxiter'] = maxiter
+
+    optimizer = optimizer(**options) #pass the arguments
+    
+    bestx = pl.numpy.array(x0, requires_grad=True)
+    besty = cost(x0, *args)
+    funcalls = 1  # tracks no. of function evals.
+    niter = 0
+    improved = True
+    stop = False  
+
+    testx = np.copy(bestx)
+    testy = np.real(besty)
+    while improved and not stop and niter < maxiter:
+        improved = False
+
+        # compute step
+        if qfim:    #natural_grad_descent
```

Comment:
Here we have an if/elif set of conditions, but they are all on different clauses:

if qfim==true --> do something
elif jac==true --> do something else

I feel all these should be plain if, and not an if/else statements.

Am I missing something obvious?